src/commands/WatchCommand.js

```diff
@@ -62,10 +62,9 @@ export default class WatchCommand extends Command {
       console.log(`${chalk.cyan(`${pkg}/${srcFile}`)} changed, running ${chalk.yellow(this.script)} script`);
       NpmUtilities.runScriptInDir(this.script, [srcFile].concat(this.args), packageLocation, (err, stdout) => {
         if (err) {
-          watcher.close();
-          callback(err);
```

Comment:
That's the callback for the watch command's `execute` function. Invoking it does cause the main process to end. Just tested that.